### PR TITLE
Kube Play: use passthrough as the default log-driver if service-container is set

### PR DESCRIPTION
**Pull request by ygalblum**
If log-driver is not set, set to passthough is service-container is set otherwise use the global default

Update the system tests
- explicitly set logdriver in sdnotify and play tests
- podman-kube template test:  Verify the default log driver for service-container

#### Does this PR introduce a user-facing change?
Yes

```release-note
When using --service-container the default log-driver is now passthrough
```

Resolves: #16592

### CODE REVIEW

File: cmd/podman/kube/play.go
Comment (rhatdan) [inline cmd/podman/kube/play.go:250]: I think a better check would be whether the user overrode the default or not.

Something like:
```
if !cmd.Flags().Changed("log-driver")  && playOptions.ServiceContainer {			
     playOptions.LogDriver = define.PassthroughLogging
}
```

**Comment by TomSweeneyRedHat** [comment]
LGTM once @rhatdan 's comment is addressed.

File: cmd/podman/kube/play.go
Comment (ygalblum) [reply cmd/podman/kube/play.go]: I didn't know this capability of the `Flags`, thanks. Fixed

**Review by vrothberg — COMMENTED** [review]
I miss an explanation on _why_. Could you add a code comment and something to the commit message on why the defaults are changed?

Also: Why only for `kube play`?  Wouldn't it make sense to cover all systemd cases?

**Comment by ygalblum** [comment]
@vrothberg 
> I miss an explanation on _why_. Could you add a code comment and something to the commit message on why the defaults are changed?

The _Why_ is explained in the issue linked to this PR. I can add a comment copying the explanation given there.
 
> Also: Why only for `kube play`? Wouldn't it make sense to cover all systemd cases?

I made the change only for `kube play` because this is what the issue referenced to. Is there an equivalent flag for `podman run`? I've just did a quick look but couldn't find something that stood out

**Comment by edsantiago** [comment]
>The Why is explained in the issue linked to this PR. I can add a comment copying the explanation given there.

Teachable moment: this is an antipattern. Linking to issues or external resources causes cognitive load on maintenance programmers. Any time a reviewer asks me "why", that's an opportunity to reflect on my code and realize that future maintainers will also wonder, and will appreciate a comment.

**Comment by ygalblum** [comment]
I've added the comment in both the code and the commit message as @vrothberg asked.
@edsantiago I asked again merely because the maintainer, (@vrothberg) who asked why, is the same one who suggested this solution :)

**Review by vrothberg — APPROVED** [review]
LGTM, thanks!

**Comment by vrothberg** [comment]
> I made the change only for kube play because this is what the issue referenced to. Is there an equivalent flag for podman run? I've just did a quick look but couldn't find something that stood out

An alternative that would also work for ordinary containers is to check whether the `PODMAN_SYSTEMD_UNIT` flag is set but it's perfectly fine as is.

> I asked again merely because the maintainer, (@vrothberg) who asked why, is the same one who suggested this solution :)

I forgot over the break that I suggested it :rofl:

**Pull request merged**
